pkg/scheduler/partition.go

```diff
@@ -1070,44 +1070,32 @@ func (pc *PartitionContext) GetRejectedApplications() []*objects.Application {
 	return appList
 }
 
-// getAppsByState returns a slice of applicationIDs for the current applications filtered by state
-// Completed and Rejected applications are tracked in a separate map and will never be included.
-func (pc *PartitionContext) getAppsByState(state string) []string {
+func (pc *PartitionContext) getAppsByInternalState(appMap map[string]*objects.Application, state string) []string {
 	pc.RLock()
 	defer pc.RUnlock()
 	var apps []string
```

Comment:
Could we avoid returning `nil`?

Reply:
declaring ```apps := []string{}``` should always return an empty list of string in cases appMap is empty.